### DISCUSSION TITLE
Question: How to get multiple matches into one capture?

### DISCUSSION THREAD

**Discussion by kothavade** [tree-sitter/tree-sitter#2539] [Q&A] [answered]
My title might not be quite accurate to what I want as I'm new to tree-sitter, but essentially I am trying to match all parameter types in one capture, for example in the following C code:
```c
Color ColorAlpha(Color c, float n) {
	return c;
}
```
I'd like to match `Color` and `float` both into a capture, say "parameter_types". So far, the closest I have gotten is:
```clj
(function_definition
        type: (_) @return_type
        declarator: (function_declarator
            declarator: (identifier) @function_name
            parameters: (parameter_list
                (parameter_declaration
                    type: (_)* @parameter_type
                )
            )
        )
    )
)
```

This will return 2 matches, one with:
```
return_type: Color
function_name: ColorAlpha
parameter_type: Color
```
and the other with:
```
return_type: Color
function_name: ColorAlpha
parameter_type: float
```

Is there a way for me to write a query where only a single match is made, with both of the parameter types in one capture? Currently I get all the matches, and then merge them by name and return type, but I was wondering if there was a simpler way to do this. So far, I've tried various usages of `*` and the "Grouping Sibling Nodes" feature described in the docs, but have not been successful with either.

**Comment by ahlinc** [comment, marked as answer]
Fix the query like the next:
```clj
(function_definition
    type: (_) @return_type
    declarator: (function_declarator
        declarator: (identifier) @function_name
        parameters: (parameter_list
            (
                (parameter_declaration
                    type: _ @parameter_type
                )
                ","?
            )*
        )
    )
)
```

![Screenshot from 2023-08-20 02-46-29](https://github.com/tree-sitter/tree-sitter/assets/14666676/faffe125-ccfb-4175-aa92-da863cfe4514)

**Reply by kothavade** [reply]
Thank you! (forgot about discussions, apologies for the unnecessary issue)
